A41-xds-rbac.md

```diff
@@ -0,0 +1,199 @@
+A41: xDS RBAC Support
+----
+* Author(s): [Eric Anderson](https://github.com/ejona86), [Ashitha
+  Santhosh](https://github.com/ashithasantosh) (RBAC engine behavior)
+* Approver: markdroth
+* Status: Draft {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2021-05-07
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Support the [xDS RBAC HTTP filter][RBAC filter] for service- and method-scoped
+client authorization (authz) on xDS-enabled gRPC servers.
+
+[RBAC filter]: https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/filters/http/rbac/v3/rbac.proto
+
+## Background
+
+[A29 xDS-Based Security][A29] introduced the ability to manage xDS-managed mTLS
+meshes. As standard for TLS clients, the client verifies the server's identity
+is permitted to run the service. This is normally known as "hostname
+verification", but in a mesh scenario may be known as "server authorization" as
+the server's certificate does not include the hostname but other identity
+information.
+
+While servers authenticated clients (i.e., the client's identity was verified),
+A29 did not include mechanism to limit which clients were authorized to access a
+server and for which gRPC methods (i.e., whether the client was permitted).
+
+Envoy defines the [RBAC HTTP filter][RBAC filter] that is well-suited for
+limiting which nodes can access which services within a mesh as encouraged by
+the principal of least privilege.
+
+Resources in REST are present in the HTTP path so it would seem RBAC could be
+used in non-gRPC environments as a precise service-level resource authz engine.
+However, this 1) requires another filter to perform end-user authentication (as
+TLS just provides service-level authn) and 2) does not scale as RBAC provides no
+mechanism for loading rules on-demand. Since gRPC encodes service-level
+resource identifiers in the request message, RBAC does not have access to the
```

Comment:
The concept of "resources" is fundamental to REST but not to gRPC so a gRPC request may not even use resource identifiers.

Reply:
I've added a parenthetical mentioning that resource ids may not even exist. The point wasn't to say that they exist, but if they do to say RBAC can't use them. While also at the same time saying this is only a minor loss in functionality.